### ISSUE TITLE
Option to run tests synchronously (& group by)

### ISSUE THREAD

**Issue by sebasslash** [Serdnad/yester#1]
Sometimes I want to be able to run tests synchronously, especially when tests are related to each other. For instance, say I have a test for a `followUser` endpoint and I also write a test for another endpoint: `unfollowUser`. It may be the case that `unfollowUser` gets called before `followUser` which results in an error because that `follow` DNE in the DB. Ideally, I want to first follow the user and then execute the unfollow test.

To make this cleaner, I'm proposing the ability to group tests and be able to define parameters that can be applied to all the tests in that group.

Like so:
```yaml
group1:
  mode: sync
  default:
    headers:
      Authorization: token
  tests:
    test1:
    test2:

group2:
  mode: async
  tests:
    test1:
    test2:
```
Let me know what you think.

**Comment by Serdnad** [comment]
Yester actually already supports this, more or less. You can chain requests to run sequentially by defining the `after` field, like so:
```yaml
test1:
  ...
  
test2:
  after: test1
  ...
```

One of the benefits to this over groups is that we can still easily take advantage of parallelization and keep the runner fast if, for example, multiple requests depend on a single request being run first, but not each other. Note the `after` field currently only supports one prior test, which is usually enough to chain a series of events, but support for requiring multiple tests is on the roadmap. Although the feature (and the project overall) could definitely use some more documentation, either in the README or in the examples, and documentation PRs are always welcome :).

As for being able to define shared values, I'd been planning to support this somehow, but sharing by groups is a pretty novel way of doing it. Could you open a separate issue for shared defaults, so as to not mix the two topics in this one issue?
